tools/setup/build_supported_browser_regex.js

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env node
+
+import {writeFile} from "node:fs/promises";
+import path from "node:path";
+import {fileURLToPath} from "node:url";
+
+import {getUserAgentRegex} from "browserslist-useragent-regexp";
+
+const root_dir = fileURLToPath(new URL("../..", import.meta.url));
+const output_path = path.join(root_dir, "web", "generated", "supported_browser_regex.ts");
+
+const BASELINE_QUERY = "baseline widely available";
```

Comment:
This script should query our actual `.browserslistrc` file, and we should put the baseline query there, to make sure our support claims match reality.

We need to use `baseline widely available with downstream`, so we don’t wrongly exclude browsers like Android Browser, Opera, and Samsung Internet.